packages/string-templates/src/helpers/javascript.js

```diff
@@ -48,14 +49,26 @@ module.exports.processJS = (handlebars, context) => {
   try {
     // Wrap JS in a function and immediately invoke it.
     // This is required to allow the final `return` statement to be valid.
-    const js = `(function(){${atob(handlebars)}})();`
+    const js = iifeWrapper(atob(handlebars))
 
     // Our $ context function gets a value from context.
     // We clone the context to avoid mutation in the binding affecting real
     // app context.
     const sandboxContext = {
       $: path => getContextValue(path, cloneDeep(context)),
```

Comment:
Random thought while I'm here: do we need this `cloneDeep` here? Accessing anything with `$(...)` in the browser does a full clone of the context, which feels a little wasteful. Could we not take a single deep clone and freeze it instead?

Reply:
That's actually a really good catch. I think I've always skimmed over this assuming that it will be only a single deep clone - but you're right that since that's inside the function body, it's surely doing a deep clone for every single binding access. If that's correct, this could potentially be a massive performance increase! I'll check and fix this.

Reply:
Yep this was invoking it every time. For a JS binding with multiple `$` usages, this will now really help performance. I've pushed this up.